gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py

```diff
@@ -47,13 +48,24 @@ def estimate_model_vram(model: Model) -> int:
     # https://github.com/vllm-project/vllm/blob/v0.6.1/vllm/worker/model_runner.py#L1313
     cuda_graph_size = 2 * 1024**3
     weight_size = 0
+    timeout_in_seconds = 5
     try:
         if model.source == SourceEnum.HUGGING_FACE:
-            weight_size = get_hf_model_weight_size(model.huggingface_repo_id)
+            weight_size = await asyncio.wait_for(
+                asyncio.to_thread(get_hf_model_weight_size, model.huggingface_repo_id),
+                timeout=timeout_in_seconds,
+            )
         elif model.source == SourceEnum.MODEL_SCOPE:
-            weight_size = get_ms_model_weight_size(model.model_scope_model_id)
+            weight_size = await asyncio.wait_for(
+                asyncio.to_thread(get_ms_model_weight_size, model.model_scope_model_id),
+                timeout=timeout_in_seconds,
+            )
+    except asyncio.TimeoutError:
+        raise asyncio.TimeoutError(
+            f"Timeout when getting weight size for model {model.name}"
+        )
     except Exception as e:
-        logger.warning(f"Cannot get weight size for model {model.name}: {e}")
+        raise Exception(f"Cannot get weight size for model {model.name}: {e}")
```

Comment:
Keep the warning. The estimation is not guaranteed. For exmaple, the metadata may not be available. It falls back to use 1 gpu in that case